src/app/newsletter/page.tsx

```diff
@@ -16,38 +19,60 @@ const getCurrentNewsletter = () => {
 
 // extract all newsletters
 const getAllNewsLetters = () => {
-  let newsletter_set = [];
-  let all_newsletters = [];
-  let newsletter_section: any = {};
+  let all_newsletters: NewsLetterSet[] = [];
 
   try {
     const dir = `${process.cwd()}/public/static/static/newsletters`;
     const folder = fs.readdirSync(dir);
     const getfolders = folder.slice(0, folder.length - 1);
 
+    // Sort folders by date to arrange them in descending order
+    const folderNameToDate = (folderName: string) => {
+      const [month, year] = folderName.split("_");
+      return new Date(`${month} 1, ${year}`);
+    };
+    getfolders.sort((a, b) => {
+      return Number(folderNameToDate(b)) - Number(folderNameToDate(a));
+    });
+
+    // loop through each folder and extract the newsletters
     for (let i = 0; i < getfolders.length; i++) {
+      let newsletter_set = [];
+      let newsletter_section: NewsLetterSet = {
+        year: "",
+        newsletters: [],
+      };
       const month_batch = getfolders[i];
-      newsletter_section[`year`] = month_batch.replace("_", " ");
+      newsletter_section["year"] = month_batch.replace("_", ", ");
       const json_dir = `${dir}/${month_batch}`;
       const json_path = fs.readdirSync(json_dir);
 
       for (let j = 0; j < json_path.length; j++) {
         const weekly_newsletter = json_path[j];
-
         const file_path = `month/newsletters/${month_batch}/${weekly_newsletter}`;
-        const data = fs.readFileSync(`${process.cwd()}/public/static/static/newsletters/${month_batch}/${weekly_newsletter}`, "utf-8");
+        const data = fs.readFileSync(
+          `${process.cwd()}/public/static/static/newsletters/${month_batch}/${weekly_newsletter}`,
+          "utf-8"
+        );
 
         const parsedData = JSON.parse(data) as NewsLetterDataType;
-        const get_title = parsedData.summary_of_threads_started_this_week.split(".")[0];
-
-        newsletter_set.push({ title: get_title, link: file_path });
+        const get_title =
+          parsedData.summary_of_threads_started_this_week.split(".")[0];
+        const existingNewsletter = newsletter_set.find(
```

Comment:
None blocking:

The code works without this extra check since you made the `newsletter_set` and `newsletter_section` variables local to the scope of the loop and only data created in that cycle are pushed to the `newsletter_section`. But I prefer the check since loops or data coming from the json file may appear twice.